### PR TITLE
Tweak name of `delay_*_bug` functions

### PR DESCRIPTION
**Pull request by compiler-errors**
r? @BoxyUwU 

open to alternative name suggestions

### CODE REVIEW

File: compiler/rustc_errors/src/lib.rs [hunk 1]
```diff
@@ -1005,10 +1004,9 @@ impl Handler {
         self.inner.borrow_mut().delay_span_bug(span, msg)
     }
 
-    // FIXME(eddyb) note the comment inside `impl Drop for HandlerInner`, that's
-    // where the explanation of what "good path" is (also, it should be renamed).
-    pub fn delay_good_path_bug(&self, msg: impl Into<DiagnosticMessage>) {
-        self.inner.borrow_mut().delay_good_path_bug(msg)
+    // FIXME(eddyb) note the comment inside `impl Drop for HandlerInner`.
```
Comment (compiler-errors) [inline compiler/rustc_errors/src/lib.rs:1007]: I don't know what this FIXME is actually asking for... to copy the explanation and give this function a good doc comment? I'm gonna leave it for now lol.

**Comment by matthiaskrgr** [comment]
Would it also make sense to rename "delay span bug" to something more descriptive?
delay_ice_with_span?
ice_(with_span_)if_no_fatal_error?
ice_with_span_unless_error?

**Comment by compiler-errors** [comment]
I wouldn't mind `delay_bug_unless_error`, but I also think that fn is much more obvious what it's doing.

**Comment by BoxyUwU** [comment]
`delay_bug_unless_error` and `delay_bug_unless_diagnostic` would pair nicely together :3

**Comment by compiler-errors** [comment]
That's a lot of changes... if people have strong feelings about this many call-sites being touched, I'd be fine with just landing the first commit.

:shrug: :smile:

**Comment by mejrs** [comment]
What about `ensure_compilation_failure`? That is what it *does*. "delaying a bug" is *how* it does it and if you're new to rustc you don't know what that means, so it just raises more questions like "what is a bug?" and "delay until what"?

Personally I spent a long time in the "I think I know what this function does but I'm not 100% sure" area, so I'd love a better name.

**Comment by lcnr** [comment]
i kinda like `assure_compilation_failure` :sparkles: don't have a strong opinion here otherwise

**Pull request closed without merging**